content/concepts/did-ddo.md

```diff
@@ -1,173 +1,513 @@
+Example:
+
+```json
+{
+  "files": "0x044736da6dae39889ff570c34540f24e5e084f4e5bd81eff3691b729c2dd1465ae8292fc721e9d4b1f10f56ce12036c9d149a4dab454b0795bd3ef8b7722c6001e0becdad5caeb2005859642284ef6a546c7ed76f8b350480691f0f6c6dfdda6c1e4d50ee90e83ce3cb3ca0a1a5a2544e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550b48181ec81673953d4eaa4b5f19a45c0e9db4cd9729696f16dd05e0edb460623c843a263291ebe757c1eb3435bb529cc19023e0f49db66ef781ca692655992ea2ca7351ac2882bf340c9d9cb523b0cbcd483731dc03f6251597856afa9a68a1e0da698cfc8e81824a69d92b108023666ee35de4a229ad7e1cfa9be9946db2d909735"
+}
+```
 
-- `metadata` - describing the asset
-- `access` - describing how the asset can be downloaded
-- `compute` - describing how the asset can be computed upon
+### Services
 
-Each asset has a `metadata` service and at least one other service.
+Services define the access to the asset, and each service is represented by its respective datatoken.
 
-Each service is distinguished by the `DDO.service.type` attribute.
+An asset should have at least one service to be actually accessible, but can have as many services which make sense for a specific use case.
 
-Each service has an `attributes` section holding the information related to the service. That section _must_ have a `main` sub-section, holding all the mandatory information that a service has to provide.
+| Attribute              | Type                          | Required                        | Description                                                                                                                                |
+| ---------------------- | ----------------------------- | ------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| **`type`**             | `string`                      | **✓**                           | Type of service (`access`, `compute`, `wss`, etc.                                                                                          |
+| **`name`**             | `string`                      |                                 | Service friendly name                                                                                                                      |
+| **`description`**      | `string`                      |                                 | Service description                                                                                                                        |
+| **`datatokenAddress`** | `string`                      | **✓**                           | Datatoken address                                                                                                                          |
+| **`providerEndpoint`** | `string`                      | **✓**                           | Provider endpoint URI (URI + path)                                                                                                         |
```

Comment:
Just double checking as I changed the description here. I assume `providerEndpoint` means the full endpoint for this service, probably `/initialize` or so. Correct?

Cause all the examples then just do `"providerEndpoint": "https://myprovider"`

Reply:
no, just root , as in example. The rest of the path is generated on the fly. Indeed , it can be renamed.

Reply:
renamed it to `providerUrl`although I guess most precise would be `providerHost`. But `providerUrl` should be fine, also so we remove the `uri` usage as it's almost never clear what difference between URI and URL is with web resources so better always use URL as that's what most people know.